### ISSUE TITLE
Unsupported data type of Deltalake

### ISSUE THREAD

**Issue by JozefDropco** [delta-io/connectors#422]
Hi,
I am sorry to disturb but I came along issue which I am not able to solve. DataTypeParser is throwing error and I didnt find a way how to go around this issue
```
java.lang.IllegalArgumentException: Failed to convert the JSON string 'interval year to month' to a data type.
	at io.delta.standalone.internal.util.DataTypeParser$.$anonfun$nameToType$1(DataTypeParser.scala:155)
```
Is there a way how to enrich the data types myself ? I have seen this class is copied from original spark but there I found the support implemented, am I able to help you somehow to implement this support? Maybe I will need some basic intro where it should be added

**Comment by zsxwing** [comment]
Delta doesn't support interval types. It's more than just parsing `interval year to month`. We need to store it in the parquet files and also make sure all of readers can read it. It will not be an easy change.

Do you have a usage of interval types? Most of the time, we just see people use them to do the date/time calculation but don't see people store intervals.

**Comment by JozefDropco** [comment]
hi @zsxwing,
I was talking with my colleagues which are primary users of DeltaLake as of now they dont need this support. They created example input for me to test my implementation. I had to support Parquet and DeltaLake read/write with custom object. I used
org.apache.parquet.hadoop.ParquetWriter<Record> and org.apache.parquet.hadoop.ParquetReader<Record> and for DeltaLake read CloseableIterator<RowRecord> and for wirte I used ParquetWriter again. Its quite hard to get the schema right because there is Spark schema (in metadata) and there is parquet schema. SparkSchema I didnt find a way how to parse it myself (without writing own parser). Generating schema to be compatible with spark is another story. I got a lot of exception till I got datatypes correct. But finally I can say it produces same results for my datatypes. Last thing which remains is the "scan functionality to implement".

thanks for your help I am closing this issue for now. If it will become again needed I will open new one and try to contribute
